pkg/apis/v1beta1/crd_validation_cel_test.go

```diff
@@ -94,6 +94,51 @@ var _ = Describe("CEL/Validation", func() {
 			Entry("invalid resource group name with unsupported chars 'name@with*unsupported&chars'", "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/name@with*unsupported&chars/providers/Microsoft.Network/virtualNetworks/vnet/subnets/subnet", false),
 		)
 	})
+
+	Context("ImageFamily and FIPSMode", func() {
+		DescribeTable("should only accept valid ImageFamily and FIPSMode combinations", func(imageFamily string, fipsMode string, expected bool) {
+			nodeClass := &v1beta1.AKSNodeClass{
+				ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName())},
+				Spec:       v1beta1.AKSNodeClassSpec{},
+			}
+			// allows for leaving imageFamily unset, which currently defaults to Ubuntu2204
+			if imageFamily != "" {
+				nodeClass.Spec.ImageFamily = &imageFamily
+			}
+			// allows for leaving fipsMode unset, which currently effectively behaves like setting fipsMode to Disabled
+			if fipsMode != v1beta1.FIPSUnspecified {
+				nodeClass.Spec.FIPSMode = &fipsMode
+			}
+			if expected {
+				Expect(env.Client.Create(ctx, nodeClass)).To(Succeed())
+			} else {
+				Expect(env.Client.Create(ctx, nodeClass)).ToNot(Succeed())
+			}
+		},
+			Entry("generic Ubuntu when FIPSMode is explicitly Disabled should succeed", "Ubuntu", v1beta1.FIPSDisabled, true),
```

Comment:
nit: I'd use the const values for the image family that are specified in the `pkg/apis/v1beta1/labels.go` file.

At least for the specified cases.

If you want to add an unspecified const in there too, you could do that as well.